### PR TITLE
[docs] Remove only cpu note due to gpu support for linear trees

### PR DESCRIPTION
**Pull request by JAroyan**
I updated the documentation note of 'linear_tree' due to the introduced support of 'device_type=gpu' in pull request  (https://github.com/microsoft/LightGBM/issues/6555).

### CODE REVIEW

File: docs/Parameters.rst [hunk 1]
```diff
@@ -795,7 +795,7 @@ Dataset Parameters
 
    -  it is recommended to rescale data before training so that features have similar mean and standard deviation
 
-   -  **Note**: works only with ``cpu`` device type and ``serial`` tree learner
+   -  **Note**: works only with ``serial`` tree learner
```
[the remark this answers is not in the document]
Comment (JAroyan) [reply docs/Parameters.rst]: Thank you for the clarification @StrikerRUS !
I added cpu and gpu as device type and did not added ``cuda`` as device type.  
As far as I could understand from the pull request, device type cuda is not yet supported or? 
The [parameter-generator.py](https://github.com/microsoft/LightGBM/blob/master/.ci/parameter-generator.py). run worked well!

[the remark this answers is not in the document]
Comment (StrikerRUS) [reply docs/Parameters.rst]: Thank you!

> As far as I could understand from the pull request, device type cuda is not yet supported or?

Yes, I have the same opinion.

**Review by StrikerRUS — APPROVED** [review]
LGTM!

Please agree with CLA by following instructions from https://github.com/microsoft/LightGBM/pull/6686#issuecomment-2422520931. After that we'll be able to merge your PR.

**Comment by JAroyan** [comment]
@microsoft-github-policy-service agree

**Pull request merged**